### ISSUE TITLE
Ability to moderate stores

### ISSUE THREAD

**Issue by ZenImagine** [drugan/commerce_multistore#1]
Hello,

I started to create a marketplace on my site and I would like to moderate the stores.
That is to say that when a user creates a shop, I do not want it to appear on the site and in the search engines (google, ...).
It must be unpublished or disabled.

If his shop is valid, I activate or publish it and change the role of the user to "Store Owner".

**Comment by drugan** [comment]
That's a good idea. What you think it the _Multistore owner_ role could be assigned from within **admin/commerce/config/stores** view? So, you'd have two additional actions:

_Assign Multistore owner role to the store owner_
_Remove Multistore owner role from the store owner_

So, the owner is not changed on the action. Just the access is removed from (or granted again to) them. Also, all the store products will be unpublished (or published) on the action. Note that in this set up products should be used only on one store.

Will be done soon...

**Comment by ZenImagine** [comment]
Yes it would be good.
I understood that it is a shortcut for changing roles and limiting access to store functionality.
But there is still a problem, the store will be visible on the site (by directly entering the address in the browser) and on the search engines.
When a node is not published, it is inaccessible. We should do the same with the stores.
If a store is disabled, it should also disable the products involved.
Or put an error message "You must disable products from this store before disabling this one".

I do not know in which case a product could be associated with several stores.

**Comment by drugan** [comment]
> But there is still a problem, the store will be visible on the site (by directly entering the address in the browser) and on the search engines.

The **view commerce_store** permission should not be granted neither to _Anonymous_ nor _Authenticated_ role user. Just because this information can be quite sensible. To check this go to **admin/commerce/config/store-types/MY_STORE_TYPE/edit/display** and try make visible all the fields on a store type. Everyone, a search engine bot including (anonymous user) can see this if they have the **view commerce_store** permission.

> I do not know in which case a product could be associated with several stores.

The site admin can assign any product to multiple stores. Also, _Multistore owner_ may create multiple stores and assign a product to each of their own stores. Note that this set up requires complex store resolvers creation but it will be done a bit later...

**Comment by ZenImagine** [comment]
Here is an example of a store :

https://www.s1biose.com/boutique/argasol-exemple-boutique

All information above the search field is the one entered when the store was created. I want them to be visible to all users.

Stores must have "Published" or "Unpublished" status.
And an additional fee "See all stores" and "View unpublished stores that I own".

**Comment by drugan** [comment]
Yes, I am also agree that having a store published/unpublished property will solve a lot of issues. I'll explore how to make it possible.

**Comment by ZenImagine** [comment]
To enable or disable a store, you must add rules.

1) If the store contains products, it can not be disabled. You must display a message "You must first deactivate the associated products to deactivate the store".

2) If a deactivated product is associated with a deactivated store and I want to activate this product. You must display a message "To activate this product, you must first activate the associated shop".

If the conditions are not met, the enable / disable check box is not clickable and the warning message is displayed.

I think this feature should be part of Drupal Commerce.

**Comment by ZenImagine** [comment]
I asked the question about Drupal Commerce

https://www.drupal.org/node/2921000

**Comment by drugan** [comment]
As I've said earlier basically it's doable. Yet, is not so easy.

Please, take into account that I am the same as you and should make money at first and only then do what I like (current module development).

I promise to fix this issue as soon as I have time for it. However, if you'll find a solution somewhere else then I'd be glad to commit it into the Multistore module too.
